kv/replica_slice.go

```diff
@@ -115,11 +119,13 @@ func (rs replicaSlice) SortByCommonAttributePrefix(attrs []string) int {
 			}
 		}
 		if firstNotOrdered == 0 {
-			return bucket
+			break
 		}
+		rs.randPerm(firstNotOrdered, topIndex, maxIndex)
 		topIndex = firstNotOrdered - 1
 	}
-	return len(attrs)
+	rs.randPerm(0, topIndex, maxIndex)
```

Comment:
you can remove this line by moving `rs.randPerm(firstNotOrdered, topIndex, maxIndex)` above `if firstNotOrdered == 0 {`